ribasim_qgis/core/topology.py

```diff
@@ -49,7 +49,7 @@ def collect_node_properties(
     node: QgsVectorLayer,
 ) -> tuple[NDArray[np.float64], NDArray[np.int_], dict[str, tuple[str, int]]]:
     n_node = node.featureCount()
-    node_fields = node.fields()
+    node_fields = node._fields()
```

Comment:
QGIS =/= Ribasim, so these should not be renamed (this is a different function altogether)